fog/view/server/src/fog_view_shard_responses_processor.rs

```diff
@@ -0,0 +1,236 @@
+
+        successful_response
+    }
+
+    fn create_failed_mvq_response(i: usize) -> MultiViewStoreQueryResponse {
+        let mut failed_response = MultiViewStoreQueryResponse::new();
+        let view_uri_string = format!("insecure-fog-view://node{}.test.mobilecoin.com:3225", i);
+        failed_response
+            .mut_decryption_error()
+            .set_fog_view_store_uri(view_uri_string);
+
+        failed_response
+    }
+
+    fn create_grpc_client(i: usize, logger: Logger) -> Arc<FogViewApiClient> {
+        let view_uri_string = format!("insecure-fog-view://node{}.test.mobilecoin.com:3225", i);
+        let view_uri = FogViewUri::from_str(&view_uri_string).unwrap();
+        let grpc_env = Arc::new(
+            grpcio::EnvBuilder::new()
+                .name_prefix("Main-RPC".to_string())
+                .build(),
+        );
+
+        let grpc_client = FogViewApiClient::new(
+            ChannelBuilder::default_channel_builder(grpc_env.clone())
+                .connect_to_uri(&view_uri, &logger),
+        );
+
+        Arc::new(grpc_client)
+    }
+
+    #[test_with_logger]
+    fn one_successful_response_no_shard_clients(logger: Logger) {
+        let grpc_client = create_grpc_client(0, logger.clone());
+        let successful_mvq_response = create_successful_mvq_response();
+        let clients_and_responses = vec![(grpc_client, successful_mvq_response)];
+
+        let result = process_shard_responses(clients_and_responses);
+
+        assert!(result.is_ok());
+
+        let shard_clients_for_retry = result.unwrap().shard_clients_for_retry;
+        assert!(shard_clients_for_retry.is_empty());
+    }
+
+    #[test_with_logger]
+    fn one_successful_response_no_pending_authentications(logger: Logger) {
+        let grpc_client = create_grpc_client(0, logger.clone());
+        let successful_mvq_response = create_successful_mvq_response();
+        let clients_and_responses = vec![(grpc_client, successful_mvq_response)];
+
+        let result = process_shard_responses(clients_and_responses);
+
+        assert!(result.is_ok());
+
+        let view_store_uris_for_authentication = result.unwrap().view_store_uris_for_authentication;
+        assert!(view_store_uris_for_authentication.is_empty());
+    }
+
+    #[test_with_logger]
+    fn one_successful_response_one_new_query_response(logger: Logger) {
+        let grpc_client = create_grpc_client(0, logger.clone());
+        let successful_mvq_response = create_successful_mvq_response();
+        let clients_and_responses = vec![(grpc_client, successful_mvq_response)];
+
+        let result = process_shard_responses(clients_and_responses);
+
+        assert!(result.is_ok());
+
+        let new_query_response = result.unwrap().new_query_responses;
+        assert_eq!(new_query_response.len(), 1);
+    }
+
+    #[test_with_logger]
+    fn one_failed_response_one_pending_shard_client(logger: Logger) {
+        let client_index: usize = 0;
```

Comment:
nit: don't think  the explicit `usize` is needed here
```suggestion
        let client_index = 0;
```

Reply:
Done.